annotate.el

```diff
@@ -896,5 +922,48 @@ file."
   (with-temp-file annotate-file
     (prin1 data (current-buffer))))
 
+(define-button-type 'annotate-summary-button
+  'follow-link t
+  'help-echo "Click to show")
+
+(defun annotate-summary-button-pressed (button)
+  "Callback called when an annotate-summary-button is activated"
+  (let ((buffer (find-file-other-window (button-get button 'file))))
+    (with-current-buffer buffer
+      (goto-char (button-get button 'go-to)))))
+
+(defun annotate-show-annotation-summary ()
+  "Show a summary of all the annotations in a temp buffer"
+  (interactive)
+  (cl-labels ((ellipsize (text)
```

Comment:
Why is this an inline function instead of a `defun`?

Reply:
Hi!

Well, i just thought this function could not be useful outside `annotate-show-annotation-summary`, if you prefer i can just refactor it into `annotate-ellipsize` or something like that.

Reply:
I think I'd prefer that, yes. I understand your intention, but factoring out the function would improve readability, I think.

Reply:
As you can see i have added some more local function instead! :)
My opinion is: as they are very specific to the enclosing function the use of `cl-label` allows me to write code without "pollute" the global namespace and keeping the name reasonably short (without the `annotation-` prefix), i would say this is exactly the reason the local function form exists! But using or not local functions is a matter of personal taste so: i suggest to keep the local functions here but i left this decision to you.

Bye!
C.

Reply:
I'm OK with your current solution. It does make the code somewhat cleaner.